### PR TITLE
Script to grab meta-data from Figma

### PR DESCRIPTION
**Pull request by nburka**
### Also done in this PR
 - adding tags to the modal
 - adding tags to the search
 - supporting multiple keywords out of order in the search
 - jest + yarn:test + yarn test:ci
 - dotenv-flow

### To test
 - see the changes in the `README.md` file
 - try running `yarn import-meta-data`
 - try running `yarn test`
 - try searching keywords and partial keywords out of order

### CODE REVIEW

File: src/scripts/importMetaData.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
+import * as Figma from 'figma-js';
+import { promises as fs } from 'fs';
+import startCase from 'lodash.startcase';
+import path from 'path';
+
+if (!process.env.FIGMA_FILENAME || !process.env.FIGMA_PERSONAL_ACCESS_TOKEN) {
```
Comment (nburka) [inline src/scripts/importMetaData.ts:6]: I can't figure out how to get the ENV vars working here. This script works by setting the env vars into constants manually.

**Comment by nburka** [comment]
Fixes https://github.com/resolvetosavelives/healthicons/issues/50

**Comment by nburka** [comment]
Fixes https://github.com/resolvetosavelives/healthicons/issues/54

File: src/lib/icons.tsx [hunk 2]
```diff
@@ -37,8 +39,13 @@ async function getIcons(dirName: string): Promise<Icon[]> {
   const fileNames = await fs.readdir(dirPath);
   return await Promise.all(
     fileNames.map(async (fileName) => {
+      const id = fileName.replace(/\.[^/.]+$/, '');
+
+      const md = metaData.find((m) => m.id === id);
```
Comment (nburka) [inline src/lib/icons.tsx:44]: use a better var name than `md`

File: src/scripts/importMetaData.ts [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+  console.log(
```
Comment (nburka) [inline src/scripts/importMetaData.ts:7]: is there a better way to output this that `console.log()`?

File: src/scripts/importMetaData.ts [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+    'You must have .env.local with FIGMA_PERSONAL_ACCESS_TOKEN and FIGMA_FILENAME'
+  );
+  process.exit();
```
Comment (nburka) [inline src/scripts/importMetaData.ts:10]: is this ok?

File: src/scripts/importMetaData.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+const metadataRegex = /([^/[]*) \[(.*)\]$/;
+
+const ICONS_PATH = path.join(process.cwd(), 'public', 'icons');
+
+function getMetadataFromDescription(name: string, description: string) {
+  const metaData = description.match(metadataRegex);
+  return metaData && metaData.length > 2
+    ? {
+        title: metaData[1],
+        tags: metaData[2].split(',').map((s) => s.trim())
```
Comment (nburka) [inline src/scripts/importMetaData.ts:27]: use `tag` instead of `s`

File: src/scripts/importMetaData.ts [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+    : {
+        title: startCase(name),
+        tags: []
+      };
+}
+
+client.file(figmaFilename).then(({ data }) => {
+  const metaData = [];
+
+  // look for icon components in: "export" with name "/filled/{category}/{name}_positive"
```
Comment (nburka) [inline src/scripts/importMetaData.ts:38]: remove colon

File: src/scripts/importMetaData.ts [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+  data.document.children.map((child) => {
+    if (child.type === 'CANVAS' && child.name === 'export') {
+      child.children.map((component) => {
+        if (
+          component.type === 'COMPONENT' &&
+          component.name.match(/^filled\//)
+        ) {
+          const idMatch = component.name.match(filenameRegex);
+          if (idMatch) {
+            const m = getMetadataFromDescription(
```
Comment (nburka) [inline src/scripts/importMetaData.ts:48]: better var name than `m`

File: src/scripts/importMetaData.ts (hunk 1 above)
Comment (nburka) [reply src/scripts/importMetaData.ts]: I figured it out. Combination of missing the dotenv-flow config() and a misnamed env var that took me 30 mins to find…

File: src/lib/icons.tsx [hunk 8]
```diff
@@ -37,8 +39,13 @@ async function getIcons(dirName: string): Promise<Icon[]> {
...
+
       return {
-        title: fileName.replace(/\.[^/.]+$/, ''),
+        title: md && md.title ? md.title : fileName.replace(/\.[^/.]+$/, ''),
```
Comment (ArGeoph) [inline src/lib/icons.tsx:47]: We can make it a bit more concise and clean by changing it to:
`title: md?.title || fileName.replace(/\.[^/.]+$/, ''),`

File: src/lib/icons.tsx [hunk 9]
```diff
@@ -37,8 +39,13 @@ async function getIcons(dirName: string): Promise<Icon[]> {
...
+        tags: md && md.tags ? md.tags : [],
```
Comment (ArGeoph) [inline src/lib/icons.tsx:48]: The same here we can transform it to  `tags: md?.tags || [],` to make it a bit easier to read

File: src/lib/icons.tsx (hunk 2 above)
Comment (ArGeoph) [reply src/lib/icons.tsx:44]: What if we use something like `currentFileMetaData` instead of just `md` here?

**Review by ArGeoph — COMMENTED** [review]
Really nice work! Just left a few comments

**Pull request merged**